### ISSUE TITLE
cwltoil --disableCaching currently requires a value

### ISSUE THREAD

**Issue by andreyto** [DataBiosphere/toil#2901]
It was a bool on/off switch before ("store_true"), but now it is a bool value option that has to be provided like `cwltoil --disableCaching some_value`. That breaks all existing code that wrapped invocations of `cwltoil` for HPC batch queueing systems.

**Comment by adamnovak** [comment]
I don't think this is cwltoil specific. I don't see a reason for this to not just be a flag.

**Comment by arostamianfar** [comment]
ah, this is a side-effect of https://github.com/DataBiosphere/toil/issues/2586 ... looks like it breaks if `--disableCaching` is the very last argument as it likely can't distinguish between the value for `disableCaching` and the positional CWL arg.
If it's not the last argument, then `--disableCaching` doesn't need a value (i.e. `cwltoil --disableCaching --retryCount 0 myfile.cwl myfile.yaml` works).
@andreyto Curious, why do you need to set `--disableCaching` in your case? It's actually True by default, so there is no need to set it.

**Comment by andreyto** [comment]
@arostamianfar I started setting it when it was ON by default, and the help advised to set it OFF on the cluster batch systems. It seems this setting has evolved breaking the compatibility with the previous code. Yes, now this option behaves differently depending on its position wrt other options, as you described. Which is probably not the intended design?

Looking at the #2586, it appears that the ON path is broken now and should not be used in any case? If so, it would be a better design to keep `--disableCaching` as false by default a `store_true` switch for compatibility, but make it a noop and update the help string accordingly.

**Comment by adamnovak** [comment]
Caching should work now; we re-wrote the CachingFileStore and it is
under CI test.

We could change to `--enableCaching` instead, if that simplifies the
parser, and keep `--disableCaching` as an option that does nothing if
we care about that backward compatibility.
